stl/inc/span

```diff
@@ -452,14 +454,17 @@ public:
     template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
     constexpr span(element_type (&_Arr)[_Size]) noexcept : _Mybase(_Size), _Mydata(_Arr) {}
 
-    template <size_t _Size, enable_if_t<_Extent == dynamic_extent || _Extent == _Size, int> = 0>
-    constexpr span(array<value_type, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
+    template <class _OtherTy, size_t _Size,
+        enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent || _Extent == _Size>,
+                        is_convertible<_OtherTy (*)[], element_type (*)[]>>,
+            int> = 0>
+    constexpr span(array<_OtherTy, _Size>& _Arr) noexcept : _Mybase(_Size), _Mydata(_Arr.data()) {}
 
-    template <size_t _Size, size_t _Ext = _Extent,
-        enable_if_t<conjunction_v<bool_constant<_Ext == dynamic_extent || _Ext == _Size>,
-                        is_convertible<const value_type*, element_type*>>,
+    template <class _OtherTy, size_t _Size,
+        enable_if_t<conjunction_v<bool_constant<_Extent == dynamic_extent || _Extent == _Size>,
+                        is_convertible<const _OtherTy (*)[], element_type (*)[]>>,

```

Comment:
`_Ext` was here to make `_Ext == dynamic_extent || _Ext == _Size` value-dependent. `_Extent == dynamic_extent || _Extent == _Size` is notably not dependent on a parameter of this template, which makes this constructor template ill-formed no diagnostic required when e.g. `_Extent == size_t{42}` since this template has no valid specializations.

Please bring it back =)

Reply:
But it is dependent on `_Size`, which is a parameter of that template. Nevertheless this is nearly dead code so...

Reply:
Also note that the constructor above has the same setup and does not use `_Ext` That was the reason I actually deleted the additional parameter

Reply:
Can you confirm that it is indeed incorrect to write it that way and why the non const array constructor did work?

Reply:
I agree with @miscco, I believe that `_Ext` should be unnecessary. If `_Extent` is 42, then this constructor can still participate if the array `_Size` is 42 (and the convertibility requirement is met), so the compiler can't complain.

Reply:
Yip, sorry - my mistake.